graphql_compiler/cost_estimation/interval.py

```diff
@@ -0,0 +1,74 @@
+import datetime
+from typing import Generic, Optional, TypeVar
+
+
+IntervalDomain = TypeVar("IntervalDomain", int, str, datetime.date, datetime.datetime)
+
+
+class Interval(Generic[IntervalDomain]):
```

Comment:
Consider making this a dataclass, and also consider making it hashable.

Reply:
Don't need to hash it yet. It's not trivial since all empty intervals are equal.